packages/private/cloudflare/index.ts

```diff
@@ -45,25 +46,31 @@ export async function publishPackage({
   name: string
   version: string
 }) {
+  const semanticVersion = semver.parse(version)
+
+  if (semanticVersion === null) throw new Error(`illegal version ${version}`)
+
   await Promise.all(
-    getEnvironments().map((env) =>
+    Array.from(getEnvironments(semanticVersion)).map((env) =>
       setCloudflarePackageValue({
         key: `${name}@${env}`,
         value: `${name}@${version}`,
       })
     )
   )
 
-  function getEnvironments() {
-    const [major, minor, patch, patchMinor] = version.split('.')
-
-    if (!patch) throw new Error(`illegal version number "${version}"`)
-
-    return [
-      ...(!patch.includes('-') ? [major, `${major}.${minor}`] : []),
-      ...(patchMinor ? [`${major}.${minor}.${patch}`] : []),
-      version,
-    ]
+  function* getEnvironments(version: semver.SemVer) {
+    const { major, minor, patch, prerelease } = version
+
+    if (!prerelease) {
+      yield `${major}`
+      yield `${major}.${minor}`
+      yield `${major}.${minor}.${patch}`
+    } else {
+      for (let i = 1; i <= prerelease.length; i++) {
+        yield `${major}.${minor}.${patch}-${prerelease.slice(0, i).join('.')}`
+      }
+    }
```

Comment:
?

Reply:
Why do we need this special syntax? Super hard to read. Could return an array instead.

Reply:
Have replaced it with a function returning an array. However coming from python the yield statement is quite natural to me :smile: